extensions/wikia/AdEngine/js/ml/hivi/leaderboardInputParser.js

```diff
@@ -0,0 +1,63 @@
+/*global define*/
+define('ext.wikia.adEngine.ml.hivi.leaderboardInputParser', [
+	'ext.wikia.adEngine.adLogicPageParams',
+	'ext.wikia.adEngine.lookup.services',
+	'wikia.log'
+], function (pageParams, lookupServices, log) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.ml.hivi.leaderboardInputParser',
+		slotName = 'TOP_LEADERBOARD';
+
+	function getBidderPrice(bidderName, prices) {
+		var price = prices[bidderName];
+
+		return price || 0;
+	}
+
+	function getData() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 19 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)